src/notation/view/abstractnotationpaintview.cpp

```diff
@@ -946,91 +917,59 @@ bool AbstractNotationPaintView::adjustCanvasPosition(const RectF& logicRect, boo
+    return moveCanvasToPosition(pos.x(), pos.y());
 }
 
 bool AbstractNotationPaintView::adjustCanvasPositionSmoothPan(const RectF& cursorRect)
 {
     RectF viewRect = viewport();
-    PointF pos(cursorRect.x() - (viewRect.width() / 2), viewRect.y());
-
-    if (!viewport().intersects(cursorRect)) {
-        pos.setY(cursorRect.y() - (viewRect.height() / 2));
-    }
-
-    return moveCanvasToPosition(pos);
+    qreal newX = cursorRect.x() - (viewRect.width() / 2);
+    qreal newY = viewRect.intersects(cursorRect)
+                 ? cursorRect.y() - (viewRect.height() / 2)
+                 : viewRect.y();
+    return moveCanvasToPosition(newX, newY);
 }
 
 bool AbstractNotationPaintView::ensureViewportInsideScrollableArea()
 {
     TRACEFUNC;
-
-    if (!m_notation) {
-        return false;
-    }
-
-    auto [dx, dy] = constraintCanvas(0, 0);
-    if (qFuzzyIsNull(dx) && qFuzzyIsNull(dy)) {
-        return false;
-    }
-
-    Transform oldMatrix = m_matrix;
-    m_matrix.translate(dx, dy);
-    onMatrixChanged(oldMatrix, m_matrix, false);
-    return true;
+    return moveCanvas(0, 0, false);
 }
 
-bool AbstractNotationPaintView::moveCanvasToPosition(const PointF& logicPos)
+bool AbstractNotationPaintView::moveCanvasToPosition(
+    qreal x,
+    qreal y,
+    bool userTriggeredMove,
+    bool overrideZoomType)
 {
     TRACEFUNC;
 
-    PointF viewTopLeft = viewportTopLeft();
-    Transform oldMatrix = m_matrix;
-    if (doMoveCanvas(viewTopLeft.x() - logicPos.x(), viewTopLeft.y() - logicPos.y())) {
-        onMatrixChanged(oldMatrix, m_matrix, false);
-        return true;
+    if (!m_notation) {
+        return false;
     }
 
-    return false;
-}
-
-bool AbstractNotationPaintView::moveCanvas(qreal dx, qreal dy)
-{
-    TRACEFUNC;
+    PointF delta = constrainedCanvasMoveDelta(x, y);
+    if (qFuzzyIsNull(delta.x()) && qFuzzyIsNull(delta.y())) {
+        return false;
+    }
 
     Transform oldMatrix = m_matrix;
-    bool moved = doMoveCanvas(dx, dy);
-
-    if (moved) {
-        onMatrixChanged(oldMatrix, m_matrix, false);
+    m_matrix.translate(delta.x(), delta.y());
+    onMatrixChanged(oldMatrix, m_matrix, overrideZoomType);
 
+    if (userTriggeredMove) {
         m_autoScrollEnabled = false;
         m_enableAutoScrollTimer.start(2000);
-
         hideElementPopup();
     }
 
-    return moved;
+    return true;
 }
 
-bool AbstractNotationPaintView::doMoveCanvas(qreal dx, qreal dy)
+bool AbstractNotationPaintView::moveCanvas(qreal dx, qreal dy, bool userTriggeredMove, bool overrideZoomType)
 {
-    if (!m_notation) {
-        return false;
-    }
-
-    if (qFuzzyIsNull(dx) && qFuzzyIsNull(dy)) {
-        return false;
-    }
-
-    auto [correctedDX, correctedDY] = constraintCanvas(dx, dy);
-    if (qFuzzyIsNull(correctedDX) && qFuzzyIsNull(correctedDY)) {
-        return false;
-    }
-
-    m_matrix.translate(correctedDX, correctedDY);
-
-    return true;
+    PointF delta = this->viewport().topLeft() - PointF(dx, dy);
```

Comment:
```suggestion
    PointF delta = viewportTopLeft() - PointF(dx, dy);
```
is slightly cheaper

(also, constructing a PointF with the only purpose of calling its `x` and `y` is a bit wasteful, but let's hope that compilers optimise it)